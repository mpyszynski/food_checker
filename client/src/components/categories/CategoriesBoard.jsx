import React from 'react';
import CategoriesBox from './CategoriesBox';
import './Categories.css';
import { Grid } from 'semantic-ui-react';

export default function CategoriesBoard({ categories }) {
  const renderCategories = categories.map(x => (
    <CategoriesBox key={x.name} name={x.name} products={x.products} id={x.id} />
  ));
  return (
    <Grid className='categories__main'>
      <Grid.Row stretched>{renderCategories}</Grid.Row>
    </Grid>
  );
}
