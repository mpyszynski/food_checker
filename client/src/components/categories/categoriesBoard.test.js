import React from 'react'
import { shallow } from 'enzyme'
import CategoriesBoard from './CategoriesBoard'

const categories = [
    {
      products: 61,
      name: 'en:Beverages',
    },
    {
      products: 52,
      name: 'en:Groceries',
    },
    {
      name: 'en:Salts',
      products: 42,
    },
    {
      name: 'en:Ketchup',
      products: 31,
    },
  ];


describe('<CategoriesBoard />', () => {
    test('should render 4 components', () => {
        const wrapper = shallow(<CategoriesBoard categories={categories} />)
        expect(wrapper.find('.categories__main').children().length).toEqual(4)
        expect(wrapper).toMatchSnapshot()
    })
    
})
