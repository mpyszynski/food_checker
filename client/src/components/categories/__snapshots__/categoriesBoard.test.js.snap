// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CategoriesBoard /> should render 4 components 1`] = `ShallowWrapper {}`;
